src/main/java/mekanism/common/PacketHandler.java

```diff
@@ -207,7 +207,7 @@ else if(data instanceof byte[])
 				}
 				else if(data instanceof ArrayList)
 				{
-					encode(((ArrayList)data).toArray(), output);
+					encode(((ArrayList<?>)data).toArray(), output);
```

Comment:
I don't see what benefit all these `<?>` calls add...

Reply:
It lets the compiler know that we explicitly don't know what the type parameter is, so it knows we're using the type safely and not doing anything with it that might unexpectedly throw a ClassCastException. (If you enable warnings for raw types and unchecked casts, you'll see them go down a lot after this change.)

Reply:
In this particular case though, IDEA at least doesn't seem to care - we're not even doing it 'safely' tbh

Reply:
Eclipse and javac both do care when the right set of warnings are on, and technically this way is safe (at least locally).